src/data_source/storage/sql.rs

```diff
@@ -583,8 +507,7 @@ impl PruneStorage for SqlStorage {
     /// This is because a vacuum operation may be necessary to reclaim more space.
     /// PostgreSQL already performs auto vacuuming, so we are not including it here
     /// as running a vacuum operation can be resource-intensive.
-
-    async fn prune(&mut self) -> Result<Option<u64>, QueryError> {
+    async fn prune(&self, pruner: &mut Pruner) -> anyhow::Result<Option<u64>> {
```

Comment:
This is the only mutation still defined on the data source itself, since it opens and commits transactions internally